pyccel/codegen/printing/ccode.py

```diff
@@ -1202,13 +1206,37 @@ def _print_Constant(self, expr):
     def _print_Return(self, expr):
         code = ''
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
+
+        if len(args) > 1:
+            if expr.stmt:
+                return self._print(expr.stmt)+'\n'+'return 0;'
+            return 'return 0;'
+
         if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
-        if len(args) == 1:
-            code +='return {0};'.format(self._print(args[0]))
-        elif len(args) > 1:
-            code += 'return 0;'
-        return code
+            # get Assign nodes form CodeBlock object expr.stmt
+            assigns = expr.stmt.get_attribute_nodes(Assign)
+
+            # check the Assign objects list in case of
+            # the user assigns a variable to an object contains IndexedElement object
+            if not assigns:
+                return 'return {0};'.format(self._print(args[0]))
+            last_assign = assigns[-1]
```

Comment:
Can there be more than one assign? If so how do we know that we can discard the others?

Reply:
According to the `Return` docstring `stmt` "represent assign stmts in the case of expression return", so i think it will be one assign unless you have a test case, i was thinking about multi-values return with expressions this should be done in this statement.
```python
        if len(args) > 1:
            if expr.stmt:
                return self._print(expr.stmt)+'\n'+'return 0;'
            return 'return 0;'
```

Reply:
No, I think you are right. I was just wondering whether it made more sense to do:
```
assert(len(assigns)==1)
last_assign = assigns[0]
```
rather than `assigns[-1]` as `assigns[-1]` gives the impression that there could be a few

Reply:
`assert(len(assigns)==1)` is a good idea, we can use it to make sure that `stmt` has one assign